inspirehep/modules/references/processors.py

```diff
@@ -0,0 +1,176 @@
+        try:
+            self.obj['number'] = int(number)
+        except (ValueError, TypeError):
+            pass
+
+    def set_record(self, record):
+        self.obj['record'] = record
+        self._ensure_field('curated_relation', False)
+
+    def curate(self):
+        self.obj['curated_relation'] = True
+
+    def set_texkey(self, texkey):
+        self.obj['texkey'] = texkey
+
+    def add_title(self, title):
+        self._ensure_field('titles', [])
+        self.obj['titles'].append({'value': title})
+
+    def add_misc(self, misc):
+        self._ensure_field('misc', [])
+        self.obj['misc'].append(misc)
+
+    def add_raw_reference(self, raw_reference, serialization='text'):
+        self._ensure_field('raw_reference', [])
+        self.obj['raw_reference'].append({
+            'value': raw_reference,
+            'serialization': serialization})
+
+    def set_year(self, year):
+        try:
+            year = int(year)
+        except (ValueError, TypeError):
+            return
+        if year >= 1000 and year <= 2050:
+            self._ensure_field('publication_info', {})
+            self.obj['publication_info']['year'] = year
+
+    def add_url(self, url):
+        self._ensure_field('urls', [])
+        self.obj['urls'].append({'value': url})
+
+    def add_author(self, full_name, role=None):
+        self._ensure_field('authors', [])
+        author = {} if not role else {'role': role}
+        author['full_name'] = full_name
+        self.obj['authors'].append(author)
+
+    def set_pubnote(self, pubnote):
+        """Parse pubnote and populate correct fields."""
+        if self.RE_VALID_PUBNOTE.match(pubnote):
+            values = split_pubnote(pubnote)
+            keys = (
+                'journal_title',
+                'journal_volume',
+                'page_start',
+                'page_end',
+                'artid')
+            self._ensure_field('publication_info', {})
+            for idx, key in enumerate(keys):
+                if values[idx]:
+                    self.obj['publication_info'][key] = values[idx]
+        else:
+            self.add_raw_reference(pubnote)
+
+    def set_publisher(self, publisher):
+        self._ensure_field('imprint', {})
+        self.obj['imprint']['publisher'] = publisher
+
+    def add_report_number(self, repno):
+        # For some reason we get more recall by trying the first part in
+        # splitting the report number.
+        repno = repno or ''
+        if _is_arxiv(repno):
+            self._ensure_field('arxiv_eprints', [])
+            self.obj['arxiv_eprints'].append(_normalize_arxiv(repno))
+        else:
+            self._ensure_field('publication_info', {})
+            self.obj['publication_info']['reportnumber'] = repno
+
+    def add_uid(self, uid):
+        """Add unique identifier in correct field."""
+        # We might add None values from wherever. Kill them here.
+        uid = uid or ''
+        if _is_arxiv(uid):
```

Comment:
The arXiv specific one is actually INSPIRE-only issue, but the other might be good if at least you open bug reports to idutils, so that others are aware.